dwave/optimization/src/nodes/_state.hpp

```diff
@@ -87,6 +87,14 @@ class ArrayNodeStateData : public NodeStateData {
 
     std::span<const Update> diff() const noexcept { return updates; }
 
+    // Append a new value to the buffer, tracking the addition in the diff
+    // Return whether a change was made.
+    bool emplace_back(double value) {
```

Comment:
This is for future implementations of `emplace_back` (or subclasses) that might do something smarter and only modify the buffer sometimes?

Reply:
The return value you mean? It's for consistency with the others. Honestly it's inherited from another method, we could probably remove them all.... but that would be out of scope for this PR I think.

Reply:
Oops, totally missed this is subclassing `NodeStateData`. Got it.